File: 0003/solution.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        dups = set()
        l = 0
        res = 0
        for r, v in enumerate(s):
            while v in dups:
                dups.remove(s[l])
                l += 1
            dups.add(v)
            res = max(res, r - l + 1)
        return res

File: 0003/solution.java
import java.util.Set;
import java.util.HashSet;

class Solution {
    public int lengthOfLongestSubstring(String s) {
        int res = 0;
        Set<Character> uniques = new HashSet<>();
        int l = 0;
        for (int r = 0; r < s.length(); r++) {
            while (uniques.contains(s.charAt(r))) {
                uniques.remove(s.charAt(l++));
            }
            uniques.add(s.charAt(r));
            res = Math.max(res, r - l + 1);
        }
        return res;
    }
}